lib/commands/exam.js

```diff
@@ -124,35 +135,89 @@ module.exports = TestCommand.extend({
    * @override
    */
   _generateCustomConfigs: function(commandOptions) {
-    var config = this._super._generateCustomConfigs.apply(this, arguments);
-
-    if (this.validator.shouldParallelize) {
-      // Get the testPage from the generated config or the Testem config and
-      // use it as the baseUrl for the parallelized test pages
-      var baseUrl = config.testPage || this._getTestPage(commandOptions.configFile);
-      var count = commandOptions.split;
-      var partitions = commandOptions.partition;
-
-      if (!partitions) {
-        partitions = [];
-        for (var i = 0; i < commandOptions.split; i++) {
-          partitions.push(i + 1);
-        }
-      }
+    const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const customBaseUrl = this._getCustomBaseUrl(config, commandOptions);
+    const shouldParallelize = this.validator.shouldParallelize;
+    const shouldLoadBalance = this.validator.shouldLoadBalance;
 
-      if (Array.isArray(baseUrl)) {
-        var command = this;
-        config.testPage = baseUrl.reduce(function(testPages, baseUrl) {
-          return testPages.concat(command._generateTestPages(baseUrl, partitions, count));
-        }, []);
-      } else {
-        config.testPage = this._generateTestPages(baseUrl, partitions, count);
+    if (!shouldParallelize && !shouldLoadBalance) {
+      return config;
+    }
+
+    let ids = commandOptions.partition;
+    let appendingParam = '_partition';
+
+    if (shouldParallelize && !ids) {
+      ids = [];
+      for (var i = 0; i < commandOptions.split; i++) {
+        ids.push(i + 1);
       }
+    } else if (shouldLoadBalance) {
+      config.additionalBrowserSocketEvents = this._addBrowserSocketEventsToConfig(config);
+      const browserCount = commandOptions.loadBalance;
+      appendingParam = '_browser';
+
+      ids = Array.from({length: browserCount}, (v,k) => k+1); // if browser=3 then id = [1,2, 3]
+    }
+
+    if (Array.isArray(customBaseUrl)) {
+      const command = this;
+      config.testPage = customBaseUrl.reduce(function(testPages, customBaseUrl) {
+        return testPages.concat(command._generateTestPages(customBaseUrl, appendingParam, ids));
+      }, []);
+    } else {
+      config.testPage = this._generateTestPages(customBaseUrl, appendingParam, ids);
     }
 
     return config;
   },
 
+  /**
+   * Customizes the base url by specified test splitting options - parellel or loadBalance.
+   * 
+   * @param {String} config
+   * @param {Object} commandOptions
+   * @return {Object}
+   * @example tests/index.html?hidepassed&_split=3 if parallel
+   *          tests/index.html?hidepassed&_split=3&_load-balance if load-balance
+   *          tests/index.html?hidepassed&_split=3&_load-balance&_partition=1&_partition=2 if load-balance and partitions are specified
```

Comment:
Again, I think you should ensure that using underscores is standard. If not, you should opt to go with no underscores for standardization.

Reply:
As looking at ember-cli example underscores are used for a case like `load-balance`.
https://github.com/ember-cli/ember-cli/blob/master/lib/commands/test.js